docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -1053,6 +1053,38 @@ Spring Cloud LoadBalancer lets you set `String` hints that are passed to the Loa
 You can set a default hint for all services by setting the value of the `spring.cloud.loadbalancer.hint.default` property. You can also set a specific value
 for any given service by setting the value of the `spring.cloud.loadbalancer.hint.[SERVICE_ID]` property, substituting `[SERVICE_ID]` with the correct ID of your service. If the hint is not set by the user, `default` is used.
 
+[[hints-based-loadbalancing]]
+=== Hint-Based Load-Balancing
+
+We also provide a `HintBasedServiceInstanceListSupplier`, which is an out-of-the-box implementation for hint-based instance selection.
+
+`HintBasedServiceInstanceListSupplier` will check for a hint request header (the default header-name is `X-SC-LB-Hint`, but you can modify it by changing the value of the `spring.cloud.loadbalancer.hint-header-name` property) and, if it finds it, will use the hint value passed in the header  to filter service instances.
+
+If no hint header has been added, `HintBasedServiceInstanceListSupplier` will use <<spring-cloud-loadbalancer-hints,hint values from properties>> to filter service instances.
+
+If there's no hint set either via the header or via properties, all service instances provided by the delegate will be returned.
+
+While filtering, `HintBasedServiceInstanceListSupplier` will look for service instances that have a matching value set under the `hint` key in their `metadataMap`. If no matching instances are found, all instances provided by the delegate will be returned.
+
+You could use this sample configuration to set it up:
```

Comment:
Change "this" to "the following". (Otherwise, I think I should look up to what you just described. Then I see the colon and realize what you mean. By then, though, I've had to read the sentence twice.)